src/server/rdb_save.cc

```diff
@@ -1159,18 +1175,21 @@ error_code RdbSaver::SaveHeader(const GlobalData& glob_state) {
   return error_code{};
 }
 
-error_code RdbSaver::SaveBody(const Cancellation* cll, RdbTypeFreqMap* freq_map) {
+error_code RdbSaver::SaveBody(Context* cntx, RdbTypeFreqMap* freq_map) {
   RETURN_ON_ERR(impl_->serializer()->FlushToSink(impl_->sink()));
 
   if (save_mode_ == SaveMode::SUMMARY) {
     impl_->serializer()->SendFullSyncCut();
   } else {
     VLOG(1) << "SaveBody , snapshots count: " << impl_->Size();
-    error_code io_error = impl_->ConsumeChannel(cll);
+    error_code io_error = impl_->ConsumeChannel(cntx->GetCancellation());
     if (io_error) {
       LOG(ERROR) << "io error " << io_error;
       return io_error;
     }
+    if (cntx->GetError()) {
+      return cntx->GetError();
+    }
```

Comment:
I don't keep track of it anymore because I worked with replication code half a year ago last time, but it should be defined whether "foreign" errors are returned within local functions (you might return an error from some other thread). I guess it makes no difference here because the handler just stops

Reply:
It's kind of tricky here because the error is reported in the fiber that feeds the channel and there's no error reporting mechanism in the channel itself (_should_ there be one?)
But I can't let `SaveEpilog()` be executed if the snapshot exited abnormally